packages/date-picker/src/vaadin-date-picker-helper.js

```diff
@@ -59,10 +59,12 @@ export function dateEquals(date1, date2) {
  * @param {!Date} date The date to check
  * @param {Date} min Range start
  * @param {Date} max Range end
+ * @param {function(!Date): boolean} isDateDisabled Callback to check if the date is disabled
  * @return {boolean} True if the date is in the range
  */
-export function dateAllowed(date, min, max) {
-  return (!min || date >= min) && (!max || date <= max);
+export function dateAllowed(date, min, max, isDateDisabled) {
```

Comment:
This is the heavy lifting function that does all the work of deciding a date needs to be disabled. So all the components that need to disable a date end up feeding here.